streamlit/streamlit_code_backup_1_02_(10_4_22).py

```diff
@@ -0,0 +1,639 @@
+)
+# Finds out the current date
+morning_selection = day_selection + timedelta(days=1)
+
+
+# Prepares variable names, depending on selection, for later
+if sensor_selection == "Sensor 5":
+    filename = str(day_selection) + "_LENSSTSL0005"
+    filename2 = str(morning_selection) + "_LENSSTSL0005"
+    sensornumber = "05"
+    filetype = "Old"
+
+if sensor_selection == "Sensor 11":
+    filename = str(day_selection) + "_LENSSTSL0011"
+    filename2 = filename2 = str(morning_selection) + "_LENSSTSL0011"
+    sensornumber = "11"
+    filetype = "Old"
+
+if sensor_selection == "Sensor 12":
+    filename = str(day_selection) + "_LENSSTSL12"
+    filename2 = str(morning_selection) + "_LENSSTSL12"
+    sensornumber = "12"
+    filetype = "New"
+
+    # Imports files
+morning_error = False
+night_error = False
+if filetype == "New":
+    try:
+        sensor = pandas.read_csv(
+            "files/" + filename + ".txt",
+            sep=";",
+            names=[
+                "UTC",
+                "CST",
+                "Temperature",
+                "Frequency",
+                "Voltage",
+                "Sensor Number",
+            ],
+        )
+    except:
+        morning_error = True
+    try:
+        sensor2 = pandas.read_csv(
+            "files/" + filename2 + ".txt",
+            sep=";",
+            names=[
+                "UTC",
+                "CST",
+                "Temperature",
+                "Frequency",
+                "Voltage",
+                "Sensor Number",
+            ],
+        )
+    except:
+        night_error = True
+else:
+    try:
+        sensor = pandas.read_csv(
+            "files/" + filename + ".txt",
+            names=["Time", "Voltage", "Frequency", "Temperature", "Sensors"],
+        )
+    except:
+        morning_error = True
+    try:
+        sensor2 = pandas.read_csv(
+            "files/" + filename2 + ".txt",
+            names=["Time", "Voltage", "Frequency", "Temperature", "Sensors"],
+        )
+    except:
+        night_error = True
+
+    # Feedback/Error messages if file is unavailable
+if morning_error == True:
+    st.error("Missing File For The Morning!")
+    if night_error == True:
+        st.error("Missing File For The Night!")
+        st.stop()
+if night_error == True:
+    st.error("Missing File For The Night!")
+if night_error == True or morning_error == True:
+    st.stop()
+
+# Sets up more variable names for later
+Frequency = sensor["Frequency"].to_list()
+Frequency2 = sensor2["Frequency"].to_list()
+Temperature = sensor["Temperature"].to_list()
+Temperature2 = sensor2["Temperature"].to_list()
+
+
+# Finds the last 0 in the first file (for the night)
+last0 = len(Frequency) - 1
+while Frequency[last0] != 0:
+    last0 = last0 - 1
+
+
+# In the event that there are zeros elsewhere in the data (ex. S5 2022-08-08), moves it back 2 hours and restarts the process
+if last0 > len(Frequency[: len(Frequency) - 120]):
```

Comment:
120 should be turned into a named variable, since it's not entirely clear what it is. 

https://www.inf.unibz.it/~calvanese/teaching/05-06-ip/lecture-notes/uni04/node17.html#:~:text=A%20magic%20number%20is%20a,is%20the%20meaning%20of%2020000%3F